test-tap/hooks.js

```diff
@@ -566,3 +567,37 @@ test('teardowns cannot be used in hooks', async t => {
 	t.ok(hookFailure);
 	t.match(hookFailure.err.message, /not allowed in hooks/);
 });
+
+test('snapshots cannot be used in hooks', async t => {
+	let hookFailure = null;
+	await promiseEnd(new Runner(), runner => {
+		runner.on('stateChange', evt => {
+			if (evt.type === 'hook-failed') {
+				hookFailure = evt;
+			}
+		});
+
+		runner.chain.before(a => {
+			a.snapshot({});
+		});
+
+		runner.chain('test', a => a.pass());
+	});
+
+	t.ok(hookFailure);
+	t.match(hookFailure.err.message, /not allowed in hooks/);
+});
+
+test('`t.try()` must not be called from hooks', async t => {
+	const ava = newAva();
+	const instance = ava.hook(async t => {
+		await t.try(tt => tt.pass());
+
+		t.pass();
+	});
+
+	const result = await instance.run();
+
+	t.notOk(result.passed);
+	t.match(result.error.savedError.message, /not allowed in hooks/);
+});
```

Comment:
I'd like to self-host these tests. See https://github.com/avajs/ava/tree/master/test/

I'm thinking a directory for "hooks" and then two tests in there? Have a look at the `test-timeouts` in https://github.com/avajs/ava/pull/2524/files.

Reply:
Addressed. I still use same naming convention like `test-timeouts` for the directory, but we can change it if you prefer `hooks`.

Reply:
Addressed. I still use same naming convention like test-timeouts for the directory, but we can change it if you prefer hooks.